salt/pin.sls

```diff
@@ -1,7 +1,7 @@
 {%- set tplroot = tpldir.split('/')[0] %}
 {%- from tplroot ~ "/map.jinja" import salt_settings with context %}
 
-{% if salt_settings.pin_version and salt_settings.version and grains.os_family|lower == 'debian' %}
+{% if grains.os_family|lower == 'debian' %}
```

Comment:
You must not remove those conditions, we don't want to render and save pin-file if `pin_verion: false` or if no `version` provided.

You just need to create separate `if` block for RedHat family. Or even better you can install `yum-plugin-versionlock` in separate state file something like `prepare.sls`.